qa/rpc-tests/test.py

```diff
@@ -0,0 +1,235 @@
+from subprocess import Popen, PIPE
+
+logging.basicConfig(format='%(asctime)s - PID:%(process)d - %(levelname)s: %(message)s', level=logging.INFO)
+
+#get system information to setup according work directory for RPCs
+HOST_OS = platform.system()
+
+#get host platform path for current user
+WORK_DIR = os.path.expanduser('~')
+
+if HOST_OS == 'Windows':
+    os.environ["BITCOINCLI"] = WORK_DIR + r"\\Documents\\zcash\\zcash-cli"
+    os.environ["BITCOIND"] = WORK_DIR + r"\\Documents\\zcash\\zcashd"
+elif HOST_OS == 'Linux':
+    os.environ["BITCOINCLI"] = WORK_DIR + r"\zcash-cli"
+    os.environ["BITCOIND"] = WORK_DIR + r"\zcashd"
+elif HOST_OS == 'Darwin':
+    os.environ["BITCOINCLI"] = WORK_DIR + r"\zcash-cli"
+    os.environ["BITCOIND"] = WORK_DIR + r"\zcashd"
+else:
+    logging.error(" %s is not currently supported.", HOST_OS)
+
+
+BASE_SCRIPTS=(
+    'paymentdisclosure.py',
+    'prioritisetransaction.py',
+    'wallet_treestate.py',
+    'wallet_anchorfork.py',
+    'wallet_changeindicator.py',
+    'wallet_import_export.py',
+    'wallet_protectcoinbase.py',
+    'wallet_shieldcoinbase_sprout.py',
+    'wallet_shieldcoinbase_sapling.py',
+    'wallet_listreceived.py',
+    'wallet.py',
+    'wallet_overwintertx.py',
+    'wallet_persistence.py', 
+    'wallet_nullifiers.py',
+    'wallet_1941.py',
+    'wallet_addresses.py',
+    'wallet_sapling.py',
+    'wallet_listnotes.py',
+    'mergetoaddress_sprout.py',
+    'mergetoaddress_sapling.py',
+    'listtransactions.py',
+    'mempool_resurrect_test.py',
+    'txn_doublespend.py',
+    'txn_doublespend.py --mineblock',
+    'getchaintips.py',
+    'rawtransactions.py',
+    'rest.py',
+    'mempool_spendcoinbase.py',
+    'mempool_reorg.py',
+    'mempool_tx_input_limit.py',
+    'mempool_nu_activation.py',
+    'mempool_tx_expiry.py',
+    'httpbasics.py',
+    'zapwallettxes.py',
+    'proxy_test.py',
+    'merkle_blocks.py',
+    'fundrawtransaction.py',
+    'signrawtransactions.py',
+    'signrawtransaction_offline.py',
+    'walletbackup.py',
+    'key_import_export.py',
+    'nodehandling.py',
+    'reindex.py',
+    'decodescript.py',
+    'blockchain.py',
+    'disablewallet.py',
+    'zcjoinsplit.py',
+    'zcjoinsplitdoublespend.py',
+    'zkey_import_export.py',
+    'reorg_limit.py',
+    'getblocktemplate.py',
+    'bip65-cltv-p2p.py',
+    'bipdersig-p2p.py',
+    'p2p_nu_peer_management.py',
+    'rewind_index.py',
+    'p2p_txexpiry_dos.py',
+    'p2p_txexpiringsoon.py',
+    'p2p_node_bloom.py',
+    'regtest_signrawtransaction.py',
+    'finalsaplingroot.py'
+)
+
+EXTENDED_SCRIPTS=(
+    'getblocktemplate_longpoll.py',
+    'getblocktemplate_proposals.py',
+    'pruning.py',
+    'forknotify.py',
+    'hardforkdetection.py',
+    'invalidateblock.py',
+    'keypool.py',
+    'receivedby.py',
+    'rpcbind_test.py',
+#   'script_test.py',
+    'smartfees.py',
+    'maxblocksinflight.py',
+    'invalidblockrequest.py',
+#    'forknotify.py',
+    'p2p-acceptblock.py'
+)
```

Comment:
I have not checked that these two lists contain all the current RPC tests. I did note that this runner does not yet support conditionally running the ZMQ or Proton RPC tests.